mitodl/mitxpro
# Adds database support for tax identifiers

#### Pre-Flight checklist


#### What are the relevant tickets?

Closes mitodl/hq#2794

#### What's this PR do?

This PR contains only model additions. 

To the `TaxRate` table:
- Adds a tax ID and tax ID name field to store MIT's tax identifier for a given country. These default to empty string and "GSTIN", respectively. 

To the `Orders` table:
- Adds a foreign key to the `TaxRate` table, solely for linking MIT's tax identifier to the order. This defaults to null. 

To the `B2BOrders` table:
- Adds the tax calculation fields - country, rate, tax name - so assessed tax (where applicable) can be tracked as it is in the Order table. 
- Adds a foreign key to the `TaxRate` table, solely for linking MIT's tax identifier to the order. This defaults to null. 
- Adds a set of fields to store the customer's tax identifier information. These default to null. 

These should facilitate the process of adding the GSTIN and collecting tax IDs for B2B orders in the frontend. They should appear in the serializers as expected. This PR doesn't contain any frontend changes - those will be taken care of in a separate issue and PR (since I believe we're waiting on that info at the moment). 

#### How should this be manually tested?

Automated tests should pass. This is just a set of model changes so you should see the new fields in the database.

#### What GIF best describes this PR or how it makes you feel?

💸
